src/linear_combination/mod.rs

```diff
@@ -96,6 +156,18 @@ fn quintic_s_box(cs: &mut dyn ConstraintSystem, lc: LinearCombination) -> Linear
   quintic.into()
 }
 
+/// Performs the Poseidon-252 hash algorithm over a set of inputs. 
+/// 
+/// In this implementation, apply the hash is the same as applying
```

Comment:
Can we reword this?

Maybe:

"Applying the hash function is the same as applying one permutation over a vector of inputs of size `width` where the first element is set to zero, and any necessary padding is added to  ensure that the input is of size `width`."

Reply:
BTW: it is possible to have the sponge construction and the merle tree having different widths and therefore having different message capacities.

Reply:
Yeah, I agree with that. I'll replace the second paragraph for yours.
Thanks!!